src/main.cpp

```diff
@@ -3753,9 +3753,14 @@ bool ProcessNewBlock(CValidationState& state, const CChainParams& chainparams, C
         CBlockIndex *pindex = NULL;
         bool fNewBlock = false;
         bool ret = AcceptBlock(*pblock, state, chainparams, &pindex, fRequested, dbp, &fNewBlock);
-        if (pindex && pfrom) {
-            mapBlockSource[pindex->GetBlockHash()] = pfrom->GetId();
-            if (fNewBlock) pfrom->nLastBlockTime = GetTime();
+        if (pfrom) {
+            if (pindex) {
+                LogPrint("block", "recv block %s (%d) size=%d peer=%d\n", pblock->GetHash().ToString(), pindex->nHeight,
+                    pfrom->nBlockSize, pfrom->id);
+                mapBlockSource[pindex->GetBlockHash()] = pfrom->GetId();
+                if (fNewBlock) pfrom->nLastBlockTime = GetTime();
+            } else
+                LogPrint("block", "recv block %s size=%d peer=%d\n", pblock->GetHash().ToString(), pfrom->nBlockSize, pfrom->id);
```

Comment:
Which is more useful to report? the height or the weight? (I am not actually sure what the weight even is!)

Shall I close this PR until I've made the changes or is it preferable to leave it open? Or shall I change the title to include WIP since it's still WIP?